### DISCUSSION TITLE
[@types/react] does not provide a JSX namespace inside react module

### DISCUSSION THREAD

**Discussion by sanchit3008** [DefinitelyTyped/DefinitelyTyped#67868] [Issues with a @types package]
After upgrading to types/styled-components 5.1.34, custom styled components are breaking as they expect JSX to be inside react module, but the types/react package is still exporting it as a global and not as a part of the react module.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
The latest release of `@types/react` for the major version you need should have it. What does `yarn why @types/react` or `npm ls @types/react` say for you?

**Reply by sanchit3008** [reply]
Hi, I am on version 17.0.2 of types/react as I am using react 17 as well. Using types/react 18+ (which is the latest version) causes react itself to stop working due to the breaking changes introduced by react 18.

Is it safe to conclude new versions of @types/styled are incompatible with react <18?

**Reply by eps1lon** [reply]
The latest 17.x release also has `React.JSX`. Same for 16.x and 15.x. You shouldn't need to upgrade the major version.

**Reply by sanchit3008** [reply]
I just tried this with the latest 17.x version, but still getting the same error until I manually redeclare the JSX namespace inside a new block called `declare module 'react'`. What am I missing here?

**Reply by eps1lon** [reply]
Are you sure you have a single version of `@types/react` installed? Maybe https://solverfox.dev/writing/duplicate-types-react helps giving you instructions to get a single version of `@types/react@17.x` installed. Otherwise please provide a minimal reproduction showcasing the issue.
